### DISCUSSION TITLE
Version 4.1

### DISCUSSION THREAD

**Discussion by Alexigbokwe** [Alexigbokwe/ExpressWebJs#54] [General]
ExpressWebJs Version 4.1 features
1. Platform switch
2. Multithreading feature for running CPU-intensive tasks.
3. Rename request validation directory from Request to Validation.
4. Optimize controller methods to support both arrow and normal functions.
5. Overload repository findAll() method to accept other arguments like sorting and pagination
6. Implement the Autowire feature for property injection for services registered in the service provider
7. Implement more collection types like List, Set, LinkedList, etc
8. Implement file processor for file uploads 
9. Fix route controller mapper.
10. Fix domain controller, service, event, and listener generator command

**Comment by LonginusChizoba** [comment]
Changing the request validation directory from Request to Validation is nice. The Request directory is kinda confusing where by I think it handles request while it's for validation. But thanks for the clarity and update.

The Autowire feature is a game changer, because sometimes I just feel like doing a property injection in ExpressWebJs which wasn't possible and am left with only constructor injection. But with the Autowire feature, property injection is now possible. Nice one.
